gstreamer/src/sample.rs

```diff
@@ -36,26 +38,29 @@ impl GstRc<SampleRef> {
         buffer: Option<&Buffer>,
         caps: Option<&Caps>,
         segment: Option<&FormattedSegment<F>>,
-        info: Option<&StructureRef>,
+        info: Option<Structure>,
     ) -> Self {
         assert_initialized_main_thread!();
         unsafe {
-            let info = info.map(|i| i.as_ptr()).unwrap_or(ptr::null());
+            let info_ptr = info.as_ref().map(|i| i.as_ptr()).unwrap_or(ptr::null());
```

Comment:
`i.into_ptr()` instead of `i.as_ptr()`, and then you can remove the `mem::forget` at the bottom :)

This is annoying, for `0.11` we should do a `info.to_glib_full()` here (which does a copy) for memory-safety.

Reply:
> i.into_ptr() instead of i.as_ptr(), and then you can remove the mem::forget at the bottom :)

Yes indeed, that's better!

> This is annoying, for 0.11 we should do a info.to_glib_full() here (which does a copy) for memory-safety.

Of course! I'll make another PR for 0.11.

Reply:
Thanks!